File: leetcode/0128LongestConsecutiveSequence/longest.cpp
#include <vector>
#include <unordered_set>
#include <cstdio>

class Solution {
public:
    int longestConsecutive(std::vector<int>& nums) {
        std::unordered_set<int> numsSet(nums.begin(), nums.end());
        int longest = 0; 
        for (const int& num : nums) {
            if (numsSet.find(num - 1) == numsSet.end()) {
                int nextNum = num + 1;
                int currentLength = 1;
                while (numsSet.find(nextNum) != numsSet.end()) {
                    nextNum++;
                    currentLength++;
                }
                longest = (currentLength > longest) ? currentLength : longest;
            }
        }
        return longest;
    }
};

int main() {
    Solution sol = Solution();
    std::vector<int> vec = {100,4,200,1,3,2};
    int res = sol.longestConsecutive(vec);
    printf("%d\n", res);
}

File: leetcode/0128LongestConsecutiveSequence/longest.py
from typing import List

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numsSet = set(nums)
        longest: int = 0

        for num in nums:
            if num - 1 not in numsSet:
                nextNum = num + 1
                currentLength = 1
                while nextNum in numsSet:
                    nextNum += 1
                    currentLength += 1
                longest = max(longest, currentLength)

        return longest
      

if __name__ == "__main__":
	sol = Solution()
	input: List[int] = [100,4,200,1,3,2]
	res = sol.longestConsecutive(input)
	print(res)